htdocs/core/class/html.formfile.class.php

```diff
@@ -689,17 +689,22 @@ function showdocuments($modulepart,$modulesubdir,$filedir,$urlsource,$genallowed
      *	@param	string	$modulepart		propal, facture, facture_fourn, ...
      *	@param	string	$modulesubdir	Sub-directory to scan (Example: '0/1/10', 'FA/DD/MM/YY/9999'). Use '' if file is not into subdir of module.
      *	@param	string	$filedir		Directory to scan
-     *  @param	string	$filter			Filter filenames on this regex string (Example: '\.pdf$')
+     *  @param	string	$filter			Filter filenames on this regex string (Example: '\.pdf$').
      *	@return	string              	Output string with HTML link of documents (might be empty string). This also fill the array ->infofiles
      */
-    function getDocumentsLink($modulepart, $modulesubdir, $filedir, $filter='')
+    function getDocumentsLink($modulepart, $modulesubdir, $filedir, $filter = null)
     {
     	include_once DOL_DOCUMENT_ROOT.'/core/lib/files.lib.php';
 
     	$out='';
     	$this->infofiles=array('nboffiles'=>0,'extensions'=>array(),'files'=>array());
 
-		$file_list=dol_dir_list($filedir, 'files', 0, preg_quote(basename($modulesubdir),'/').'[^\-]+', '\.meta$|\.png$');	// Get list of files starting with name fo ref (but not followed by "-" to discard uploaded files)
+    	if (!$filter) {
+		    // Get list of files starting with name fo ref (but not followed by "-" to discard uploaded files)
+    		$filter = preg_quote(basename($modulesubdir),'/').'[^\-]+(odt|pdf)$';
```

Comment:
The submodule to generate documents may generate other files than .odt and .pdf but thay can also generate .txt, .odx, etc...
Any document must be returned, except ".meta" that is reserved to store a meta description file and images use to save thumbnails.